x/stake/keeper/delegation.go

```diff
@@ -380,15 +380,15 @@ func (k Keeper) Delegate(ctx sdk.Context, delAddr sdk.AccAddress, bondAmt sdk.Co
 func (k Keeper) unbond(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress,
 	shares sdk.Dec) (amount sdk.Dec, err sdk.Error) {
 
-	k.OnDelegationSharesModified(ctx, delAddr, valAddr)
-
 	// check if delegation has any shares in it unbond
 	delegation, found := k.GetDelegation(ctx, delAddr, valAddr)
 	if !found {
 		err = types.ErrNoDelegatorForAddress(k.Codespace())
 		return
 	}
 
+	k.OnDelegationSharesModified(ctx, delAddr, valAddr)
```

Comment:
Huh it is not immediately obvious why this needed to be moved. Can you add a comment or something?

Reply:
hmm going to disagree on this one. I think this should stay without a comment, or rather the relevant comment is the one that already exists above `check if delegation has any shares in it unbond`
the the delegation doesn't exist nothing can proceed (will cause panics in the hook)

Reply:
I see, :+1: